bitcoind-rpc/src/main/scala/org/bitcoins/rpc/serializers/JsonSerializers.scala

```diff
@@ -1,386 +1,425 @@
+
+  implicit val unspentOutputReads: Reads[UnspentOutput] =
+    Json.reads[UnspentOutput]
+
+  // Other Models
+  implicit val blockTransactionReads: Reads[BlockTransaction] =
+    Json.reads[BlockTransaction]
+  implicit val getBlockTemplateResultReads: Reads[GetBlockTemplateResult] =
+    Json.reads[GetBlockTemplateResult]
+
+  implicit val miningInfoReads: Reads[GetMiningInfoResult] =
+    Json.reads[GetMiningInfoResult]
+
+  implicit val memoryManagerReads: Reads[MemoryManager] =
+    Json.reads[MemoryManager]
+  implicit val getMemoryInfoResultReads: Reads[GetMemoryInfoResult] =
+    Json.reads[GetMemoryInfoResult]
+
+  implicit val validateAddressResultReads: Reads[ValidateAddressResultImpl] =
+    Json.reads[ValidateAddressResultImpl]
+
+  implicit val embeddedResultReads: Reads[EmbeddedResult] =
+    Json.reads[EmbeddedResult]
+
+  implicit val addressInfoResultReads: Reads[AddressInfoResult] =
+    Json.reads[AddressInfoResult]
+
+  implicit val receivedLabelReads: Reads[ReceivedLabel] =
+    Json.reads[ReceivedLabel]
+
+  implicit val estimateSmartFeeResultReads: Reads[EstimateSmartFeeResult] =
+    Json.reads[EstimateSmartFeeResult]
+
+  implicit val walletProcessPsbtResultReads: Reads[WalletProcessPsbtResult] =
+    Json.reads[WalletProcessPsbtResult]
+
+  implicit val finalizedPsbtReads: Reads[FinalizedPsbt] = FinalizedPsbtReads
+
+  implicit val nonFinalizedPsbtReads: Reads[NonFinalizedPsbt] =
+    NonFinalizedPsbtReads
+
+  implicit val finalizePsbtResultReads: Reads[FinalizePsbtResult] =
+    FinalizePsbtResultReads
+
+  implicit val rpcPsbtOutputReads: Reads[RpcPsbtOutput] = RpcPsbtOutputReads
+
+  implicit val psbtBIP32DerivsReads: Reads[PsbtBIP32Deriv] =
+    PsbtBIP32DerivsReads
+
+  implicit val rpcPsbtScriptReads: Reads[RpcPsbtScript] = RpcPsbtScriptReads
+
+  implicit val psbtWitnessUtxoInputReads: Reads[PsbtWitnessUtxoInput] =
+    Json.reads[PsbtWitnessUtxoInput]
+
+  implicit val mapPubKeySignatureReads: Reads[
+    Map[ECPublicKey, ECDigitalSignature]] = MapPubKeySignatureReads
+
+  implicit val rpcPsbtInputReads: Reads[RpcPsbtInput] = RpcPsbtInputReads
+
+  implicit val decodePsbtResultReads: Reads[DecodePsbtResult] =
+    Json.reads[DecodePsbtResult]
+
+  implicit val psbtMissingDataReads: Reads[PsbtMissingData] =
+    Json.reads[PsbtMissingData]
+
+  implicit val analyzePsbtInputReads: Reads[AnalyzePsbtInput] =
+    Json.reads[AnalyzePsbtInput]
+
+  implicit val analyzePsbtResultReads: Reads[AnalyzePsbtResult] =
+    Json.reads[AnalyzePsbtResult]
+
+  implicit val getNodeAddressesReads: Reads[GetNodeAddressesResult] = { js =>
+    for {
+      time <- js.validate[Long].map(FiniteDuration(_, TimeUnit.SECONDS))
+      services <- js.validate[Int]
+      address <- js.validate[URI]
+      port <- js.validate[Int]
+    } yield GetNodeAddressesResult(time, services, address, port)
+  }
+
+  implicit val RpcCommandsReads: Reads[RpcCommands] =
+    Json.reads[RpcCommands]

```

Comment:
This line throws a compile error, because we have no reader for `FiniteDuration`. Implementing something similar to `getNodeAddressesReads` would fix this